Java/core/src/main/java/com/amazon/randomcutforest/tree/RandomCutTree.java

```diff
@@ -306,6 +306,19 @@ protected void increaseMassOfAncestors(Node mergedNode) {
         }
     }
 
+    @Override
+    protected void decreaseMassOfAncestors(Node mergedNode) {
+        Node parent = mergedNode.getParent();
+        while (parent != null) {
+            parent.decrementMass();
+            parent = parent.getParent();
+        }
+    }
+
+    @Override
+    void setLeafPointReference(Node node, double[] point) {
```

Comment:
This is interesting, because there's a sensible implementation of this method (namely, set `node.leafPoint = point`) but this method isn't being invoked for non-compact trees.